x-pack/plugin/stack/src/test/java/org/elasticsearch/xpack/stack/StackTemplateRegistryTests.java

```diff
@@ -538,6 +540,11 @@ public void testThatTemplatesAreNotDeprecated() {
             .forEach(p -> assertFalse((Boolean) p.get("deprecated")));
     }
 
+    public void testDataStreamLifecycleNodeFeatureId() {
+        // let's make sure these ids remain in-sync
+        assertThat(StackTemplateRegistry.SUPPORTS_DATA_STREAM_LIFECYCLE.id(), is(DataStreamFeatures.SUPPORTS_DATA_STREAM_LIFECYCLE.id()));
```

Comment:
And also another spot
```suggestion
        assertThat(StackTemplateRegistry.DATA_STREAM_LIFECYCLE.id(), is(DataStreamFeatures.DATA_STREAM_LIFECYCLE.id()));
```